### ISSUE TITLE
Template menu: Fix semantics of the Preview template menu item

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#57455]
### Description

See https://github.com/WordPress/gutenberg/pull/56817

The 'Template preview' menu item in the template manu has:
- `role="menuitem"`
- `aria-pressed` true or false 

This is incorrect. The [aria-pressed](https://www.w3.org/TR/wai-aria-1.2/#aria-pressed) state can only be used with the `role=button` (native buttons or elements with that ARIA role).

It's not just a matter of forma compliance. Screen readers will not announce the 'selected' state of the menu item. This should be instead:
- role="menuitemcheckbox"
- `aria-checked` true or false 

It's no different from what already done for other dropdown with `role="menu"`, for example the Options nenu:
- A role="menu" should only contain menuitem, menuitemcheckbox, menuitemradio, and optionally groups.
- Normal menu items have a `menuitem` role.
- Menu items with a checkable state must either have a [menuitemcheckbox](https://www.w3.org/TR/wai-aria-1.2/#menuitemcheckbox) or a [menuitemradio](https://www.w3.org/TR/wai-aria-1.2/#menuitemradio) role.
- For both menuitemcheckbox and menuitemradio roles, `aria-checked` is a required state.

### Step-by-step reproduction instructions

- Test with Safari and VoiceOver.
- Tab to the template menu in the Document settings and open it.
- Tab to the 'Template preview' menu item and press Control+Option+Spacebar to check it and uncheck it.
- Observe that in both checked and unchecked states, the state is not announced.

Regardless of the visual checked state and the (incorrect) aria-pressed value, VoiveOver always announce the item as 'menu item' and does not announce the state. Screenshot:

![Screenshot 2023-12-29 at 14 36 26](https://github.com/WordPress/gutenberg/assets/1682452/6c03f14c-0375-4b29-b3f8-2b4a797971e5)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
One more way to test:
- Install the axe devtools browser extension, available for Firefox and Chrome.
- Open the template menu.
- Run the axe checker from your browser dev tools.
- Observe one of the error reported is:

`ARIA attribute is not allowed: aria-pressed="true"`

Hint: Press the 'Highlight' button in the axe devtools to highlight the element with error in the UI.

Screenshot:

![Screenshot 2023-12-29 at 14 19 25](https://github.com/WordPress/gutenberg/assets/1682452/d7bf7614-f89e-4661-8ad5-922b0b663c55)
